File: C++/ArrayString/05_majorityElement.cpp
#include <bits/stdc++.h>
using namespace std;

int majorityElement(vector<int> &nums)
{
    int ans = 0, count = 0;
    for (int i = 0; i < nums.size(); i++)
    {
        if (count == 0)
            ans = nums[i];
        if (nums[i] == ans)
            count++;
        else
            count--;
    }
    return ans;
}

File: Python/ArrayString/05_majorityElement.py
def majorityElement(nums):

    nums.sort()
    count, ans = 1, 0
    
    for i in range(1, len(nums)):
        if(nums[i] == nums[i - 1]):
            count += 1;
        else:
            count = 1
        if(count > len(nums) // 2):
            return nums[i - 1]
    
    return nums[0]